ipe_process_orchestrator/assign_competencies.py

```diff
@@ -52,8 +52,10 @@ def get_student_list_with_course_grades(self) -> List[Dict[str, Any]]:
             logging.debug(f'Params for next page: {page_info}')
             student_list_payload = page_info
             page_num += 1
-      logger.info(f'{len(students_with_grades)} active enrolled students may receive IPE competencies in the course {self.course[COL_COURSE_ID]}')
-      return students_with_grades
+      
+      student_grade_with_out_duplicates = pd.DataFrame(students_with_grades).drop_duplicates(subset='student_canvas_id').to_dict('records')
+      logger.info(f"""{len(student_grade_with_out_duplicates)} active enrolled students may receive IPE competencies in the course {self.course[COL_COURSE_ID]}{'. 0 students received competencies.' if len(student_grade_with_out_duplicates)==0 else '.'} """)
```

Comment:
minor wording change: student_grade_with_out_duplicates -> student_grade_without_duplicates

Reply:
if I understand logic of this line correctly, it will print

"len(student_grade_with_out_duplicates) active enrolled students may receive IPE competencies in the course xxxxx" when len(student_grade_with_out_duplicates) > 0

vs 

"0 active enrolled students may receive IPE competencies in the course xxxxx. 0 students received competencies." when  len(student_grade_with_out_duplicates) == 0

I feel it would be enough to do:

logger.info(f"""{len(student_grade_with_out_duplicates)} active enrolled students may receive IPE competencies in the course {self.course[COL_COURSE_ID]}{'. 0 students received competencies.' if len(student_grade_with_out_duplicates)==0 else '.'} """)

Reply:
OK, I changed the as per your suggestion. I took the if statement here and put it below when I an checking for no student enrollments case

Reply:
looks good!